### ISSUE TITLE
Service not being added to *.bot file

### ISSUE THREAD

**Issue by szul** [Microsoft/BotFramework-Emulator#642]
It seems to be touch and go as to whether the `*.bot` file for a bot is actually updated when adding services through the UI. In adding an Azure Bot Service, LUIS, and QnAMaker service, only the QnAMaker information was inserted into the services array in the `*.bot` file.

Also, the UI itself, on the services tab where the ribbons are shown for each service, doesn't actually show any services as being connected, until you refresh--even for the services that successfully have entries in the `*.bot` file.

It could be that there are validations occurring, but no validation errors are being presented in the UI.

**Comment by tonyanziano** [comment]
Can't repro this issue at the moment. I'll wait for the next pre-release build of the emulator and try to repro it then.

**Comment by tonyanziano** [comment]
@szul We have actually been really behind on putting out a new v4 release, so it's very possible this issue has been fixed in the latest, but has not been published in a release yet. I was building the latest locally when I tried to repro, and was unable to.

I assume you're still using v4.0.15-alpha right?

We are aiming to put out a new release (v4.0.16-alpha) soon, which should fix a lot of lingering issues.
